toolkit/tools/licensecheck/licensecheck.go

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// A tool for validating the license files of RPM packages in a set of directories.
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/exe"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/file"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/logger"
+	"github.com/microsoft/azurelinux/toolkit/tools/pkg/licensecheck"
+	"github.com/microsoft/azurelinux/toolkit/tools/pkg/licensecheck/licensecheckformat"
+
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+var (
+	app = kingpin.New("licensecheck", "A tool for validating the license files of RPM packages.")
+
+	rpmDirs       = app.Flag("rpm-dirs", "Directories to recursively scan for RPMs to validate").Required().ExistingDirs()
+	nameFile      = app.Flag("name-file", "File containing license names to check for.").Required().ExistingFile()
+	exceptionFile = app.Flag("exception-file", "File containing license exceptions.").ExistingFile()
+	mode          = app.Flag("mode", "Level of license validation to perform").Default(string(licensecheck.LicenseCheckModeDefault)).Enum(licensecheck.ValidLicenseCheckModeStrings()...)
+
+	buildDirPath = app.Flag("build-dir", "Directory to store temporary files.").Required().String()
+	distTag      = app.Flag("dist-tag", "The distribution tag.").Required().String()
+	workerTar    = app.Flag("worker-tar", "Full path to worker_chroot.tar.gz.").Required().ExistingFile()
+
+	logFlags    = exe.SetupLogFlags(app)
+	resultFile  = app.Flag("results-file", "The file to store the search result.").Default("").String()
+	summaryFile = app.Flag("summary-file", "File to save the license check summary to.").String()
+)
+
+func main() {
+	app.Version(exe.ToolkitVersion)
+	kingpin.MustParse(app.Parse(os.Args[1:]))
+	logger.InitBestEffort(logFlags)
+
+	mode := licensecheck.LicenseCheckMode(*mode)
+	if !licensecheck.IsValidLicenseCheckMode(mode) {
```

Comment:
`kingpin` has a `.Enum()` function you could use on the `Flag` if you were so inclined.

Reply:
Hmm, not sure why I added this, probably force of habit. I'm already using the enum to validate this.